### PR TITLE
Allow more master options to be configurable

### PR DESCRIPTION
**Pull request by smarterclayton**
[test]

### CODE REVIEW

File: pkg/cmd/server/api/types.go [hunk 1]
```diff
@@ -235,6 +235,11 @@ type ServingInfo struct {
 	ServerCert CertInfo
 	// ClientCA is the certificate bundle for all the signers that you'll recognize for incoming client certificates
 	ClientCA string
+	// MaxRequestsInFlight is the number of concurrent requests allowed to the server. If zero, no limit.
```
Comment (liggitt) [inline pkg/cmd/server/api/types.go]: this gets used by lots of things (etcd, master, assets, node)... probably shouldn't add options here unless we support them uniformly

Comment (smarterclayton) [inline pkg/cmd/server/api/types.go]: The alternative is to create another top level struct for them for each section that needs them.  Most Go things *would* support this, and probably should.

File: pkg/cmd/server/api/types.go [hunk 2]
```diff
@@ -235,6 +235,11 @@ type ServingInfo struct {
...
+	MaxRequestsInFlight int
+	// RequestTimeoutSeconds is the number of seconds before requests are timed out. The default is 60 minutes, if
+	// -1 there is no limit on requests.
```
Comment (liggitt) [inline pkg/cmd/server/api/types.go]: why not 0 as no limit?

Comment (liggitt) [inline pkg/cmd/server/api/types.go]: oh, default state... *sigh*

File: pkg/cmd/server/api/v1/conversions.go [hunk 3]
```diff
@@ -15,6 +15,19 @@ func init() {
...
+		},
+		func(obj *NodeConfig) {
+			if obj.ServingInfo.RequestTimeoutSeconds == 0 {
+				obj.ServingInfo.RequestTimeoutSeconds = 60 * 60
+			}
+		},
+		func(obj *EtcdConfig) {
+			if obj.ServingInfo.RequestTimeoutSeconds == 0 {
+				obj.ServingInfo.RequestTimeoutSeconds = 60 * 60
+			}
```
Comment (liggitt) [inline pkg/cmd/server/api/v1/conversions.go]: can you just add a defaulting function for `*ServingInfo`?

File: pkg/cmd/server/origin/master.go [hunk 4]
```diff
@@ -494,11 +494,22 @@ func (c *MasterConfig) Run(protected []APIInstaller, unprotected []APIInstaller)
 		handler = contextHandler
 	}
 
+	if c.Options.ServingInfo.MaxRequestsInFlight > 0 {
+		longRunningRE := regexp.MustCompile("[.*\\/watch$][^\\/proxy.*]")
```
Comment (liggitt) [inline pkg/cmd/server/origin/master.go]: what about pods/exec, pods/proxy, pods/logs, etc?

File: pkg/cmd/server/api/types.go (hunk 2 above)
Comment (smarterclayton) [inline pkg/cmd/server/api/types.go]: So we can set a sane default for existing users.

File: pkg/cmd/server/origin/master.go (hunk 4 above)
Comment (smarterclayton) [inline pkg/cmd/server/origin/master.go]: Hahahhahahaha you think the person who implemented this upstream thought of those.

File: pkg/cmd/server/api/types.go (hunk 1 above)
Comment (smarterclayton) [inline pkg/cmd/server/api/types.go]: How about HTTPServingInfo which embeds ServingInfo?

File: pkg/cmd/server/api/v1/conversions.go (hunk 3 above)
Comment (smarterclayton) [inline pkg/cmd/server/api/v1/conversions.go]: I think we want different defaults

**Comment by smarterclayton** [comment]
Added HTTPServingInfo which embeds ServingInfo PTAL

**Comment by smarterclayton** [comment]
Part of #2628

**Comment by smarterclayton** [comment]
[merge]

**Comment by smarterclayton** [comment]
Retroactively approved

**Pull request merged**